pygmy/model/link.py

```diff
@@ -62,9 +66,41 @@ def generate_short_code(_, connection, target):
                     is_default=is_default
                 )
             )
+            return short_code
+ 
+    @staticmethod
+    def generate_base64_qr_code(_, connection, target, shorted):
+        table = Link.__table__
+        if not target.qr_code:
+            # Generate qr code
+            qr = qrcode.QRCode(
+                version=10,
+                error_correction=qrcode.constants.ERROR_CORRECT_M,
+                box_size=10,
+                border=4,
+            )
+            qr.add_data("https://herme.li/" + shorted)
+            qr.make(fit=True)
+            img = qr.make_image(fill_color="black", back_color="white")
+            # Convert PIL image to bytes
+            img_byte_arr = BytesIO()
+            img.save(img_byte_arr, format='PNG')
+            img_byte_arr = img_byte_arr.getvalue()
+            # Convert bytes to base64
+            qr_code = base64.b64encode(img_byte_arr)
+            connection.execute(
```

Comment:
This should probably be wrapped in a `Try Except` statement in case the DB connection fails.